addons/web/static/src/core/emoji_picker/emoji_picker.js

```diff
@@ -248,10 +248,16 @@ export class EmojiPicker extends Component {
     }
 
     get recentEmojis() {
-        return Object.entries(this.state.recent)
+        const recent = Object.entries(this.state.recent)
             .sort(([, usage_1], [, usage_2]) => usage_2 - usage_1)
             .slice(0, 42)
             .map(([codepoints]) => this.emojiByCodepoints[codepoints]);
+        if (this.searchTerm.length > 0 && recent.length > 0) {
```

Comment:
As `searchTerm` is a string you can just check whether `searchTerm` has a Truthy value or not. No, need to check it's length.